tools/CMakeLists.txt

```diff
@@ -11,33 +11,33 @@
 # directory. Library targets and header files should be placed in the
 # appropriate subtree, e.g. `compiler/src/iree/compiler/Tools/`.
 #
-# Programs with a dependency on the compiler are not designed to run on device
-# platforms (e.g. Android), so they are tagged "hostonly".
+# Compiler tools are designed to run on host platforms (Linux, macOS, Windows),
+# so they are only built when IREE_BUILD_COMPILER is set and are tagged
+# "hostonly". When cross-compiling (or generally wanting to use already-built
+# compiler tools), set the IREE_HOST_BINARY_ROOT CMake option.
 #
 # This file does not use bazel_to_cmake because of special logic throughout.
 
-# TODO(#6353): Tools has thread dependencies in gtest, benchmark, yaml, etc.
-# This should be split between runtime/compiler with optional threading support.
-if(NOT IREE_ENABLE_THREADING)
-  return()
-endif()
-
 # Write some important CMake options to a file for convenient use from scripts.
 configure_file(build_config_template.txt.in build_config.txt)
 
-# Depending on which target backends are enabled, we may or may not have built
-# LLD.
-if(IREE_LLD_TARGET)
-  # lld install - required by the compiler to link codegen executables.
-  install(
-    TARGETS lld
-    COMPONENT Compiler
-    RUNTIME DESTINATION bin
-  )
+iree_add_all_subdirs()
+
+# TODO(scotttodd): Should this be checking IREE_BUILD_COMPILER?
+#   Maybe we should disallow setting both at the same time, since it's
+#   ambigious which should be used
+if(IREE_HOST_BINARY_ROOT AND NOT IREE_BUILD_COMPILER)
+  iree_import_binary(NAME iree-tblgen)
+  iree_import_binary(NAME iree-compile)
+  iree_import_binary(NAME iree-opt)
+  iree_import_binary(NAME iree-run-mlir)
```

Comment:
return?

Reply:
The `IREE_ENABLE_THREADING` check made the ordering in this file messier than I'd like, so this should not return. The _runtime_ tools (like `iree-benchmark-module`) are defined right after the threading check.